File: frontend/src/types/product.ts
export interface ProductImage {
    id?: number
    image_url?: string
    image_type?: string
    display_order?: number
    main?: string // Add this to match usage, or fix usage
}

// Better approach: Define the images object structure if it's not an array of ProductImage
export interface ProductImages {
    main?: string
    gallery?: string[]
}

export interface ProductVariant {
    id: number
    name: string
    sku: string
    price_adjustment: number // Added
    stock_quantity: number
    specifications?: Record<string, string | number | boolean>
}

export interface Category {
    id: number
    name: string
    slug: string
    parent_id?: number | null
    display_order?: number
    children?: Category[]
}

export interface Brand {
    id: number
    name: string
    slug: string
    logo_url?: string
    is_active?: boolean
}

export interface Product {
    id: number
    name: string
    slug: string
    sku?: string
    description?: string
    clinical_benefits?: string
    specifications?: Record<string, string | number | boolean>
    base_price: number
    selling_price: number
    stock_quantity: number // Added
    has_variants?: boolean
    is_active: boolean
    brand_id?: number
    category_id?: number

    // Relations
    brand?: Brand
    category?: Category
    images?: ProductImage[]
    variants?: ProductVariant[]
}

export interface ProductListResponse {
    items: Product[]
    total: number
    page: number
    size: number
    pages: number
}

File: backend/app/schemas/product.py
from typing import List, Optional, Dict, Any
from pydantic import BaseModel

class ProductImageBase(BaseModel):
    image_url: str
    image_type: str
    display_order: int

    class Config:
        from_attributes = True

class Brand(BaseModel):
    id: int
    name: str
    slug: str
    logo_url: Optional[str] = None
    is_active: bool = True

    class Config:
        from_attributes = True

class ProductVariantBase(BaseModel):
    sku: str
    name: str
    price: float
    stock_quantity: int
    specifications: Optional[Dict[str, Any]] = None

    class Config:
        from_attributes = True

class ProductVariant(ProductVariantBase):
    id: int

class CategoryBase(BaseModel):
    id: int
    name: str
    slug: str
    display_order: int = 0
    parent_id: Optional[int] = None

    class Config:
        from_attributes = True

class ProductBase(BaseModel):
    name: str
    slug: str
    brand_id: Optional[int] = None
    category_id: Optional[int] = None
    description: Optional[str] = None
    base_price: float
    selling_price: float
    has_variants: bool = False
    is_active: bool = True

class Product(ProductBase):
    id: int
    images: List[ProductImageBase] = []
    variants: List[ProductVariant] = []

    class Config:
        from_attributes = True

class ProductCreate(ProductBase):
    pass

class ProductUpdate(ProductBase):
    pass
